src/std_extensions/collections.rs

```diff
@@ -96,35 +103,31 @@ impl ConstFold for PopFold {
         let list: &ListValue = list.get_custom_value().expect("Should be list value.");
         let mut list = list.clone();
         let elem = list.0.pop()?; // empty list fails to evaluate "pop"
-        let list = make_list_const(list, type_args);
+        let list = make_list_const(list);
         let elem = ops::Const::new(elem, ty.clone()).unwrap();
 
         Some(vec![(0.into(), list), (1.into(), elem)])
     }
 }
 
-pub(crate) fn make_list_const(list: ListValue, type_args: &[TypeArg]) -> ops::Const {
-    let list_type_def = EXTENSION.get_type(&LIST_TYPENAME).unwrap();
-    ops::Const::new(
-        list.into(),
-        Type::new_extension(list_type_def.instantiate(type_args).unwrap()),
-    )
-    .unwrap()
+pub(crate) fn make_list_const(list: ListValue) -> ops::Const {
```

Comment:
I think you can change the impl block in src/ops/constant.rs:
```
impl<T> From<T> for Const
where
    T: KnownTypeConst + CustomConst,
```
to just `impl <T: CustomConst> From<T> for Const`, at which point this `make_list_const` is just `Const::from` so can be removed